.github/workflows/build.yml

```diff
@@ -20,33 +24,35 @@ jobs:
       - uses: gradle/wrapper-validation-action@v1
       - uses: gradle/gradle-build-action@v2
         with:
-          arguments: |
-            build
-            projectHealth
-            mergeLintSarif
-            mergeDetektSarif
-            :plugins:buildPlugins
-            --continue
-      # checks integrity of release tasks, update publish workflows too if does not work
-      - uses: gradle/gradle-build-action@v2
-        with:
-          arguments: |
-            publishAppyxReleasePublicationToOSSRHRepository
-            publishAppyxReleasePublicationToSonatypeSnapshotRepository
-            --dry-run
-            --no-parallel
-      - uses: gradle/gradle-build-action@v2
-        if: github.ref == 'refs/heads/main' && github.repository == 'bumble-tech/appyx'
+          cache-read-only: ${{ env.MAIN_BRANCH != 'true' }}
+      - name: Build
+        run: >
+          ./gradlew
+          build
+          projectHealth
+          mergeLintSarif
+          mergeDetektSarif
+          :plugins:buildPlugins
+          --continue
+      - name: Check publication setup
+        run: >
+          ./gradlew
+          publishAppyxReleasePublicationToOSSRHRepository
+          publishAppyxReleasePublicationToSonatypeSnapshotRepository
+          --dry-run
+          --no-parallel
+      - name: Deploy snapshot
+        if: env.MAIN_BRANCH == 'true' && github.repository == 'bumble-tech/appyx'
```

Comment:
This could be problematic, the 1.0 and 2.0 snapshots will override each other. Perhaps we need to also change the snapshot version to handle 1.x and 2.x branches?

Reply:
Will split.